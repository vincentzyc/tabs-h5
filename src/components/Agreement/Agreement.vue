<template>
  <van-checkbox v-model="checked" icon-size="14px" checked-color="#666666" label-disabled>
    <p class="agreement-text">
      <span>{{ title }}</span>
      <span v-for="(agr, key) in agrList">
        <span v-if="key > 0">å’Œ</span>
        <span class="agreement-title" @click="openAgreement(agr.title, agr.text)">{{ agr.title }}</span>
      </span>
    </p>
  </van-checkbox>
  <AgreementPopup v-model:show="showPopup" :text="curText" />
</template>

<script setup lang="ts">
defineProps<{
  title: string;
  agrList: {
    title: string;
    text: string;
  }[];
}>();
let checked = $ref(false),
  showPopup = $ref(false),
  curText = $ref("");

function openAgreement(title: string, text: string) {
  showPopup = true;
  curText = text;
}
</script>
