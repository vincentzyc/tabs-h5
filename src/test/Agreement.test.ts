import { mount } from '@vue/test-utils'
import Agreement from '@/components/Agreement/Agreement.vue'

test('mount Agreement component', async () => {
  expect(Agreement).toBeTruthy()

  const wrapper = mount(Agreement, {
    props: {
      title: "填写并提交视为阅读并同意",
      checked: false,
      agrList: [
        {
          title: "《个人信息保护政策》",
          text: "https://h5.lipush.com/h5/index.html?id=5523261663728019648",
        },
        {
          title: "《入网许可协议》",
          text: "https://h5.lipush.com/h5/index.html?id=5220700664176004699",
        },
        {
          title: "《单独同意书》",
          text: "https://h5.lipush.com/h5/index.html?id=7883210563728253535",
        },
      ],
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
  const listLength = wrapper.props().agrList.length

  expect(wrapper.findAll('.agreement-title')).toHaveLength(listLength)
})
