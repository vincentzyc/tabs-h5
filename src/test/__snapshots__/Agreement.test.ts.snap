// Vitest Snapshot v1

exports[`mount Agreement component 1`] = `
"<div role=\\"checkbox\\" class=\\"van-checkbox van-checkbox--label-disabled\\" tabindex=\\"0\\" aria-checked=\\"false\\">
  <div class=\\"van-checkbox__icon van-checkbox__icon--round\\" style=\\"font-size: 14px;\\"><i class=\\"van-badge__wrapper van-icon van-icon-success\\">
      <!---->
      <!---->
      <!---->
    </i></div><span class=\\"van-checkbox__label\\"><p class=\\"agreement-text\\"><span>填写并提交视为阅读并同意</span><span><!--v-if--><span class=\\"agreement-title\\">《个人信息保护政策》</span></span><span><span>和</span><span class=\\"agreement-title\\">《入网许可协议》</span></span><span><span>和</span><span class=\\"agreement-title\\">《单独同意书》</span></span></p></span>
</div>
<!--teleport start-->
<!--teleport end-->"
`;
